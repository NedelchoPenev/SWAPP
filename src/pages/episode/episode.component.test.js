import React from 'react';
import { shallow } from 'enzyme';

import EpisodePage from './episode.component';
import CustomButton from '../../components/custom-button/custom-button.component';

describe('EpisodePage component', () => {
  const mockId = 'films.4';
  const mockTitle = 'The Phantom Menace';
  const mockOpeningCrawl = 'Test test test';
  const imageUrl = 'www.testImage.com';
  const mockDirector = 'George Lucas';
  const mockReleaseDate = '1999-05-19';
  const mockPeopleHasNextPage = {
    edges: [
      {
        node: {
          id: 'test',
        },
      },
    ],
    pageInfo: {
      hasNextPage: true,
    },
  };
  const mockPeopleNONextPage = {
    edges: [
      {
        node: {
          id: 'test',
        },
      },
    ],
    pageInfo: {
      hasNextPage: false,
    },
  };

  it('should render EpisodePage component', () => {
    const mockProps = {
      episode: {
        episodeId: mockId,
        title: mockTitle,
        openingCrawl: mockOpeningCrawl,
        image: imageUrl,
        director: mockDirector,
        releaseDate: mockReleaseDate,
        people: mockPeopleHasNextPage,
      },
    };

    expect(shallow(<EpisodePage {...mockProps} />)).toMatchSnapshot();
  });

  it('should NOT render button component if hasNextPage is false', () => {
    const mockProps = {
      episode: {
        episodeId: mockId,
        title: mockTitle,
        openingCrawl: mockOpeningCrawl,
        image: imageUrl,
        director: mockDirector,
        releaseDate: mockReleaseDate,
        people: mockPeopleNONextPage,
      },
    };

    expect(shallow(<EpisodePage {...mockProps} />).find(CustomButton)).toEqual(
      {},
    );
  });
});
