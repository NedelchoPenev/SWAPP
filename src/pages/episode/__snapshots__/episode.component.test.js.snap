// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage component should render EpisodePage component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="The Phantom Menace"
      src="www.testImage.com"
    />
    <styled.div>
      <h1>
        Star Wars: Episode 
      </h1>
      <h2>
        The Phantom Menace
      </h2>
    </styled.div>
  </styled.div>
  <styled.div>
    <p>
      Test test test
    </p>
    <p>
      <span
        id="pointer"
      >
        Director:
      </span>
       
      <span
        id="content"
      >
        George Lucas
      </span>
    </p>
    <p>
      <span
        id="pointer"
      >
        Release Date: 
      </span>
      <span
        id="content"
      >
        1999-05-19
      </span>
    </p>
  </styled.div>
  <styled.div>
    <styled.div>
      <withRouter(CharacterCard)
        character={
          Object {
            "id": "test",
          }
        }
        key="test"
      />
    </styled.div>
    <CustomButton>
       Load More 
    </CustomButton>
  </styled.div>
</styled.div>
`;
